### ISSUE TITLE
List shows matches instead of captured group

### ISSUE THREAD

**Issue by justhalf** [gskinner/regexr#152]
For some cases, putting "$1\n" in the "List" tab below displays the whole match instead of capturing group 1.

Example: http://regexr.com/3de84

When I hover over the text above, it clearly shows (correctly) that the capturing group 1 doesn't contain any comma, but in the "List", when I try to display "$1\n", it displays the comma.

**Comment by gskinner** [comment]
Good catch! We have to do some fancy slicing and dicing of strings to make List work, and that seems to be causing some problems with the start of string token (^).

Here's a simplified reproduction case:
http://regexr.com/3deun

I'll try to take a look at this and figure out a solution soon, but feel free to dig into the code if you're interested.
